diff --git b/exp/bbob_test_01.m a/exp/bbob_test_01.m
index 5a3f06f..5689f7a 100644
--- b/exp/bbob_test_01.m
+++ a/exp/bbob_test_01.m
@@ -52,7 +52,7 @@ function bbob_test_01(id, exp_id, exppath_short, varargin)
   instances = bbParams.instances;
   maxfunevals = bbParams.maxfunevals;
 
-  % try
+  try
 
   for dim = bbParams.dimensions            % small dimensions first, for CPU reasons
     % for ifun = benchmarks('FunctionIndices')  % or benchmarksnoisy(...)
@@ -133,8 +133,7 @@ function bbob_test_01(id, exp_id, exppath_short, varargin)
     fprintf('---- dimension %d-D done ----\n', dim);
   end
 
-  return;
-  % catch err
+  catch err
     save([resultsFile '_ERROR.mat']);
     fprintf('#########################################################\n');
     fprintf('#########################################################\n');
@@ -155,7 +154,7 @@ function bbob_test_01(id, exp_id, exppath_short, varargin)
     % whole Matlab if an error occures
     exit(1);
     throw(err);
-  % end
+  end
 end
 
 function [exp_results, tmpFile, cmaes_out] = runTestsForAllInstances(opt_function, id, exp_settings, datapath, opt, maxrestarts, maxfunevals, minfunevals, t0, exppath, localDatapath)
